py-polars/polars/internals/construction.py

```diff
@@ -1229,15 +1229,40 @@ def to_frame_chunk(
     return (df.rechunk() if n_chunks > 0 else df)._df
 
 
+def pandas_has_default_index(df: pd.DataFrame) -> bool:
+    """Identify if the pandas frame only has a default (or equivalent) index."""
+    from pandas.core.indexes.numeric import IntegerIndex
+    from pandas.core.indexes.range import RangeIndex
+
+    index_cols = df.index.names
+
+    if len(index_cols) > 1:
+        return False  # not default: more than one index
+    elif index_cols not in ([None], [""]):
+        return False  # not default: index is named
+    elif df.index.equals(RangeIndex(start=0, stop=len(df), step=1)):
+        return True  # is default: simple range index
+    else:
+        # finally, is the index _equivalent_ to a default unnamed
+        # integer index with frame data that was previously sorted
+        return isinstance(df.index, IntegerIndex) and sorted(
```

Comment:
[Pandas recommends using to_numpy()](https://pandas.pydata.org/docs/reference/api/pandas.Index.values.html) instead of values. Also, because the pandas index is a numpy array under the hood, wouldn't it be slightly better to use `(df.index.sort_values() == np.arange(len(df))).all()` without having to convert to a python lits?

Reply:
Actually I think `.array` would be better here, but good catch ;)